src/main/java/org/jenkinsci/plugins/workflow/job/properties/PipelineTriggersJobProperty.java

```diff
@@ -0,0 +1,176 @@
+    private List<Trigger<?>> triggers = new ArrayList<>();
+
+    @DataBoundConstructor
+    public PipelineTriggersJobProperty(List<Trigger<?>> triggers) {
+        // Defensive handling of when we get called via {@code Descriptor.newInstance} with no form data.
+        if (triggers == null) {
+            this.triggers = new ArrayList<>();
+        } else {
+            this.triggers = triggers;
+        }
+    }
+
+    public void setTriggers(List<Trigger<?>> triggers) {
+        this.triggers = triggers;
+    }
+
+    public List<Trigger<?>> getTriggers() {
+        return triggers;
+    }
+
+    public void addTrigger(Trigger t) {
+        triggers.add(t);
+    }
+
+    public Trigger getTriggerForDescriptor(TriggerDescriptor td) {
+        for (Trigger trigger : triggers) {
+            if (td.equals(trigger.getDescriptor())) {
+                return trigger;
+            }
+        }
+
+        return null;
+    }
+
+    public void stopTriggers() {
+        for (Trigger trigger : triggers) {
+            trigger.stop();
+        }
+    }
+
+    public void startTriggers(boolean newInstance) {
+        for (Trigger trigger : triggers) {
+            trigger.start(owner, newInstance);
+        }
+    }
+
+    public void removeTrigger(Trigger t) {
+        // TODO: Will we get equality for trigger instances of the same Descriptor?
+        Trigger toRemove = getTriggerForDescriptor(t.getDescriptor());
+
+        if (toRemove != null) {
+            triggers.remove(toRemove);
+        }
+    }
+
+    public Map<TriggerDescriptor,Trigger<?>> getTriggersMap() {
+        Map<TriggerDescriptor,Trigger<?>> triggerMap = new HashMap<>();
+
+        for (Trigger t : getTriggers()) {
+            TriggerDescriptor td = t.getDescriptor();
+            triggerMap.put(td, t);
+        }
+
+        return triggerMap;
+    }
+
+    public List<Action> getAllTriggerActions() {
+        List<Action> triggerActions = new ArrayList<>();
+
+        for (Trigger<?> t : triggers) {
+            triggerActions.addAll(t.getProjectActions());
+        }
+
+        return triggerActions;
+    }
+
+    @CheckForNull
+    @Override
+    public PipelineTriggersJobProperty reconfigure(@Nonnull StaplerRequest req, @CheckForNull JSONObject form) throws Descriptor.FormException {
+        DescribableList<Trigger<?>,TriggerDescriptor> trigList = new DescribableList<>(Saveable.NOOP);
+        try {
+            trigList.rebuild(req, form, Trigger.for_(owner));
+        } catch (IOException e) {
+            // TODO: Not sure what form field would make sense here?
+            throw new Descriptor.FormException(e, null);
+        }
+
+        this.stopTriggers();
+
+        PipelineTriggersJobProperty thisProp = new PipelineTriggersJobProperty(new ArrayList<>(trigList.toList()));
+        thisProp.setOwner(owner);
+
+        thisProp.startTriggers(true);
+
+        return thisProp;
+    }
+
+    @Extension
+    @Symbol("pipelineTriggers")
```

Comment:
@kohsuke it is a pity that the symbol must be unique amongst all `JobProperty`s, even though `isApplicable` (inherited and using the `WorkflowJob` type parameter) means that for a given job or job type just `triggers` would suffice. But I guess fixing that could be complicated: `resolveClass` in `structs` would need to be supplied with a filter predicate to apply to candidate extensions.